app/controllers/dish_photo_likes_controller.rb

```diff
@@ -5,6 +5,7 @@ class DishPhotoLikesController < ApplicationController
   def create
     if @dish_photo and !current_user.likes?(@dish_photo)
       current_user.like!(@dish_photo)
+      @dish_photo.create_activity :like, owner: current_user, recipient: @dish_photo.dish.user
     end
   end
```

Comment:
Consider removing the `before_action` statement on line 3 and removing the `get_dish_photo` private method to make the `create` action easier to understand. I think the `get_dish_photo` method is adding complexity without benefit.

This would move the `@dish_photo = DishPhoto.find(...)` line into `create`. 

`Model.find(...)` raises `ActiveRecord::RecordNotFound` when no match is found. This means that there's no need for the nil check on line 6, which would become:

```ruby
if !current_user.likes?(@dish_photo)
```